src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -329,9 +329,15 @@ private void retrieveBranches(@NonNull final SCMHeadObserver observer, @NonNull
     }
 
     private void observe(SCMHeadObserver observer, final TaskListener listener,
-            final String owner, final String repositoryName, 
+            final String owner, final String repositoryName,
             final String branchName, final String hash, BitbucketPullRequest pr) throws IOException {
-        if (isExcluded(branchName)) {
+
+        String branchNameForExclusionTest = branchName;
+        if (pr != null) {
+             branchNameForExclusionTest = String.format("PR-%s %s", pr.getId(), pr.getSource().getBranch().getName());
```

Comment:
I don't think this is related to the fix explained in the description, but useful I think.